tensorflow/python/ops/clip_ops.py

```diff
@@ -212,6 +212,8 @@ def clip_by_norm(t, clip_norm, axes=None, name=None):
     values = ops.convert_to_tensor(
         t.values if isinstance(t, indexed_slices.IndexedSlices) else t,
         name="t")
+    if clip_norm < 0:
```

Comment:
This won't work if `clip_norm` is a tensor.

I think your best option is to set
```
clip_norm = math_ops.maximum(clip_norm, 0)
```
to force it to be positive.

Reply:
@cantonios 
Thanks for the inputs. Actually I have tested with a Tensor also by passing `clip_norm=tf.constant(-10.0)` and the proposed solution still working and same can be referred in the gist attached in above [comment](https://github.com/tensorflow/tensorflow/pull/61242#issue-1798546174).

I have also tested with `tf.function` and the proposed solution still working and raising intended error.Please refer [gist2](https://colab.research.google.com/gist/SuryanarayanaY/40fd4154a5835c65c5493407cd2f9ef8/61158_errors-for-code-fix-2.ipynb).

for example below code outputs bool Tensor and hence it seems working for this case:

```
clip_norm=tf.constant(-10.0)
print(clip_norm <0)

#output:tf.Tensor(True, shape=(), dtype=bool)
```
I am not sure though which cases it may fail.

Reply:
That's because you're giving it constant tensors, so in the `tf.function` can extract the constant value during tracing.  If it was a variable tensor, it would fail.

For example, the following works fine with the original `tf.clip_by_norm`:
```python
@tf.function
def tf_clip_by_norm(t, clip_norm, axes=[1]):
  return tf.clip_by_norm(t, clip_norm=clip_norm, axes=[1])

# Current tf.clip_by_norm:
clip = tf_clip_by_norm.get_concrete_function(tf.TensorSpec(shape=[3, 3], dtype=tf.float32), tf.TensorSpec(shape=[], dtype=tf.float32), tf.TensorSpec(shape=[], dtype=tf.int32))
print(clip(tf.ones([3, 3]), tf.constant(0.5), tf.constant(1)))
```

But with your modification, it incorrectly throws the `ValueError` immediately when trying to get the concrete function:
```python
clip = clip_by_norm.get_concrete_function(tf.TensorSpec(shape=[3, 3], dtype=tf.float32), tf.TensorSpec(shape=[], dtype=tf.float32), tf.TensorSpec(shape=[], dtype=tf.int32))
```
```python
ValueError: in user code:

    File "<ipython-input-9-abfa95055a32>", line 22, in clip_by_norm  *
        raise ValueError('clip_norm should be a 0-D (scalar) Tensor > 0')

    ValueError: clip_norm should be a 0-D (scalar) Tensor > 0
```
even though we haven't actually passed in any values.

Reply:
@cantonios ,

Thanks for clarification. Then I will go with converting -ve values to 0 as suggested using:

`clip_norm = math_ops.maximum(clip_norm, 0)
`

Also I would like to add a note of same in the Documentation to make the users aware of this.

Thanks

Reply:
Yes, please update the documentation.